docs/reference/aggregation.md

```diff
@@ -0,0 +1,488 @@
+---
+title: Aggregation
+sidebar_position: 11
+---
+
+An aggregation summarizes your data as statistics on buckets or metrics.
+
+Aggregations can provide answer to questions like:
+
+- What is the average price of all sold articles?
+- How many errors with status code 500 do we have per day?
+- What is the average listing price of cars grouped by color?
+
+There are two categories: [Metrics](#metric) and [Buckets](#bucket-aggregations).
+
+#### Prerequisite
+
+To be able to use aggregations on a field, the field needs have a fast field index created. A fast field index is a columnar storage, 
+where documents values are extracted and stored to.
+
+Example to create a fast field on text for term aggregations.
+```yaml
+name: category
+type: text
+tokenizer: raw
+record: basic
+fast: true
+```
+
+See the [index config](./index-config.md) for more details and examples.
+
+#### Format
+
+The aggregation request and result de/serialize into elasticsearch compatible JSON. 
+If not documented otherwise you should be able to drop in your elasticsearch aggregation queries.
+
+In some examples below is not the full request shown, but only the payload for `aggregations`.
+
+#### Example
+
+Request
+```json
+{
+  "query": "*",
+  "max_hits": 1,
+  "aggregations": {
+    "sites_and_aqi": {
+      "terms": {
+        "field": "County",
+        "size": 2,
+        "order": {"average_aqi": "asc"}
+      },
+      "aggs": {
+        "average_aqi": {
+          "avg": {
+            "field": "AQI"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+
+Response
+```json
+...
+"aggregations": {
+    "sites_and_aqi": {
+      "buckets": [
+        {
+          "average_aqi": {
+            "value": 32.62267569707098
+          },
+          "doc_count": 56845,
+          "key": "臺東縣"
+        },
+        {
+          "average_aqi": {
+            "value": 35.97893635571055
+          },
+          "doc_count": 28675,
+          "key": "花蓮縣"
+        }
+      ],
+      "sum_other_doc_count": 1872055
+    }
+```
+#### Limitations
+
+Currently aggregations work only on single value fast fields of type u64, f64, i64 and on string fields.
```

Comment:
it's `text` field with `raw` tokenizer only, we don't have yet a "string" field but that would be nice to have one (like a shortcut to define a text field with tokenizer raw).

Reply:
I like the idea of a tag field.

tag = indexed + fast + raw tokenizer.